src/api/common/schemas/amountbase.json

```diff
@@ -0,0 +1,44 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "amountbase",
+  "description": "Base class for amount and issue",
+  "type": "object",
+  "properties": {
+    "value": {
+      "description": "The quantity of the currency, denoted as a string to retain floating point precision",
+      "$ref": "value"
+    },
+    "currency": {
+      "description": "The three-character code or hex string used to denote currencies",
+      "$ref": "currency"
+    },
+    "counterparty": {
+      "description": "The Ripple account address of the currency's issuer or gateway, or an empty string if the currency is XRP",
+      "$ref": "address"
```

Comment:
The address schema doesn't allow empty strings. It might be better to use:

```json
"anyOf": [
  { "$ref": "address" },
  {"type": "string", "description": "Empty string",  "pattern": "^$" }
]
```